cmd/gb-vendor/fetch.go

```diff
@@ -56,7 +56,11 @@ Flags:
 			return err
 		}
 
-		wc, err := repo.Checkout(branch, revision)
+		if m.HasDependencyWithUrl(repo.URL()) {
+			return fmt.Errorf("specified repository is already downloaded")
```

Comment:
What if you check that both d.URL and d.Path match.

On Sun, Jun 7, 2015 at 7:52 PM, Dávid Kaya <notifications@github.com> wrote:

> In cmd/gb-vendor/fetch.go
> <https://github.com/constabulary/gb/pull/192#discussion_r31875430>:
>
> > @@ -56,7 +56,11 @@ Flags:
> >  			return err
> >  		}
> >
> > -		wc, err := repo.Checkout(branch, revision)
> > +		if m.HasDependencyWithUrl(repo.URL()) {
> > +			return fmt.Errorf("specified repository is already downloaded")
>
> Something like fmt.Errorf("%s is already vendored from %s repository",
> path, repo.URL()) ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/192/files#r31875430>.
>